src/pages/transactions/edit/Edit.tsx

```diff
@@ -115,34 +121,47 @@ export function Edit() {
 
     setErrors(undefined);
 
-    setIsSaving(true);
-
-    toast.processing();
-
-    try {
-      await request('PUT', endpoint('/api/v1/bank_transactions/:id', { id }), {
-        ...transaction,
-        amount: Number(transaction?.amount),
-        base_type:
-          transaction?.base_type === TransactionType.Deposit
-            ? ApiTransactionType.Credit
-            : ApiTransactionType.Debit,
-      });
-
-      toast.success(t('updated_transaction'));
-      setIsSaving(false);
-      navigate(route('/transactions'));
-    } catch (error) {
-      const axiosError = error as AxiosError;
-
-      console.error(axiosError);
-
-      if (axiosError?.response?.status === 422) {
-        setErrors(axiosError?.response?.data?.errors);
-        toast.dismiss();
-      } else {
-        toast.error(t('error_title'));
+    if (transaction?.bank_integration_id) {
+      setIsSaving(true);
+
+      toast.processing();
+
+      try {
+        await request(
+          'PUT',
+          endpoint('/api/v1/bank_transactions/:id', { id }),
+          {
+            ...transaction,
+            amount: Number(transaction?.amount),
+            base_type:
+              transaction?.base_type === TransactionType.Deposit
+                ? ApiTransactionType.Credit
+                : ApiTransactionType.Debit,
+          }
+        );
+
+        toast.success('updated_transaction');
+        setIsSaving(false);
+        navigate('/transactions');
+      } catch (error) {
+        const axiosError = error as AxiosError;
+
+        console.error(axiosError);
+
+        if (axiosError?.response?.status === 422) {
+          setErrors(axiosError?.response?.data?.errors);
+          toast.dismiss();
+        } else {
+          toast.error(t('error_title'));
+        }
+
+        setIsSaving(false);
       }
+    } else {
+      setErrors((prevState) => ({
+        ...prevState,
+        bank_integration_id: t('required_field'),
+      }));
     }
```

Comment:
Please use API validation, not frontend.

Reply:
replied above